File: app/src/main/java/com/prinkal/pos/app/dialog/SimpleProgressDialog.java
package com.prinkal.pos.app.dialog;

import android.app.Dialog;
import android.content.Context;
import android.view.LayoutInflater;

import com.prinkal.pos.app.databinding.DialogSimpleProgressBinding;

import androidx.annotation.NonNull;

public class SimpleProgressDialog extends Dialog {
  private Context mContext;
  
  private DialogSimpleProgressBinding binding;
  
  public SimpleProgressDialog(@NonNull Context context) {
    super(context);
    requestWindowFeature(1);
    this.binding = DialogSimpleProgressBinding.inflate(LayoutInflater.from(context), null, false);
    setContentView(this.binding.getRoot());
  }
  
  public static SimpleProgressDialog show(Context context, String title, String subTitle) {
    SimpleProgressDialog dialog = new SimpleProgressDialog(context);
    dialog.binding.heading.setText(title);
    dialog.binding.subheading.setText(subTitle);
    dialog.show();
    return dialog;
  }
}

File: app/src/main/java/com/prinkal/pos/app/dialog/SimpleProgressDialog.kt
package com.prinkal.pos.app.dialog

import android.app.Dialog
import android.content.Context
import android.view.LayoutInflater
import com.prinkal.pos.app.databinding.DialogSimpleProgressBinding

class SimpleProgressDialog(context: Context) : Dialog(context) {
    private val mContext: Context? = null
    private val binding: DialogSimpleProgressBinding

    companion object {
        fun show(context: Context, title: String?, subTitle: String?): SimpleProgressDialog {
            val dialog = SimpleProgressDialog(context)
            dialog.binding.heading.text = title
            dialog.binding.subheading.text = subTitle
            dialog.show()
            return dialog
        }
    }

    init {
        requestWindowFeature(1)
        binding = DialogSimpleProgressBinding.inflate(LayoutInflater.from(context), null, false)
        setContentView(binding.root)
    }
}